### PR TITLE
Allow other mapping formats to be used

### PR DESCRIPTION
**Pull request by Geolykt**
The "fix" is not the prettiest in the world but as far as I know there isn't really a better one without peeking at the file data.

### CODE REVIEW

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/enigma/EnigmaMappingsReader.java [hunk 1]
```diff
@@ -47,6 +47,9 @@
 	DIRECTORY {
 		@Override
 		public EntryTree<EntryMapping> read(Path root, ProgressListener progress, MappingSaveParameters saveParameters) throws IOException, MappingParseException {
+			if (!Files.isDirectory(root)) {
+				throw new MappingParseException(root::toString, 0, "Path is not a directory!");
```
Comment (2xsaiko) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/enigma/EnigmaMappingsReader.java:51]: java.nio.file.NotDirectoryException? This isn't really a parser error.

**Review by 2xsaiko — CHANGES_REQUESTED** [review]

File: enigma-cli/src/main/java/cuchaz/enigma/command/Command.java [hunk 2]
```diff
@@ -41,22 +41,28 @@ protected static EnigmaProject openProject(Path fileJarIn, Path fileMappings) th
...
+		Exception last = null;
+		if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
+			return MappingFormat.ENIGMA_ZIP.read(path, progress, saveParameters);
 		} else {
-			return MappingFormat.ENIGMA_FILE;
+			for (MappingFormat format : MappingFormat.getReadableFormats()) {
+				try {
+					return format.read(path, progress, saveParameters);
+				} catch (Exception e) {
+					last = e;
```
[the remark this answers is not in the document]
Comment (Kroppeb) [reply enigma-cli/src/main/java/cuchaz/enigma/command/Command.java:61]: You can add them as suppressed exceptions.

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java [hunk 3]
```diff
@@ -32,6 +32,13 @@ public MappingParseException(Supplier<String> filenameProvider, int line, String
 		filePath = filenameProvider.get();
 	}
 
+	public MappingParseException(Supplier<String> filenameProvider, int line, Throwable cause) {
+		super(cause);
+		this.line = line;
+		this.message = cause.toString();
+		filePath = filenameProvider.get();
```
[the remark this answers is not in the document]
Comment (Geolykt) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java:39]: It is also used by the other constructors in the class, which is why I kept it in the same style, I guess we could refractor that?

[the remark this answers is not in the document]
Comment (2xsaiko) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java:39]: Ah sorry, didn't see that. Yeah sure, refactoring both sounds like a good idea.

**Review by 2xsaiko — APPROVED** [review]

**Pull request merged**